src/domain/wallet-invoices/index.types.d.ts

```diff
@@ -8,7 +8,10 @@ type WalletInvoice = {
 
 interface IWalletInvoicesRepository {
   persist: (invoice: WalletInvoice) => Promise<WalletInvoice | RepositoryError>
+
   findByPaymentHash: (
     paymentHash: PaymentHash,
   ) => Promise<WalletInvoice | RepositoryError>
+
+  findWalletsWithPendingInvoices: () => AsyncGenerator<WalletId> | RepositoryError
```

Comment:
The prefix `find` implies a criteria we are looking up. I think better would be `list` or `iter` for this function.